### ISSUE TITLE
LB listens on all interfaces when used with scheduler IP's

### ISSUE THREAD

**Issue by aemneina** [rancher/rancher#9752]
**Rancher versions:**
rancher/server: v1.6.7

It's not feasible to specify an IP when deploying more than one LB. LB's run on multiple nodes for resilience.

It would be great if the LB behaved like other services and was assigned a scheduler IP at random, as thats how regular services work. The problem can be exacerbated when a load-balancer hijacks the traffic for another well-behaved service.

**Comment by carolynhu** [comment]
Manual tests:
1. Launch three hosts in UI:
<img width="958" alt="screen shot 2017-09-18 at 2 27 49 pm" src="https://user-images.githubusercontent.com/29208297/30565554-983a9b48-9c7d-11e7-98a6-8589774b5352.png">
2. add additional ip for hosts

- for "default" host, add two extra fake ip: 192.168.10.10, 192.168.20.20

- for "default2" host, add two extra fake ip: 192.168.30.30, 192.168.40.40

- for "default3" host, add two extra fake ip: 192.168.50.50, 192.168.60.60

3.create a "test" stack, and create a service (named "service1") in "test" stack, which being launched in "default3" host. 

4.create a load balancer service (named "lb") in "test" stack, and set port "11:11" and "22:22", choose service1 as private port's container.
which, we can see that, ip allocated to "lb" is "default2" host's ip, which is correct (since here we want to make sure the allocated ip not comes from other hosts).
<img width="553" alt="screen shot 2017-09-18 at 2 32 23 pm" src="https://user-images.githubusercontent.com/29208297/30565742-4edcbcf0-9c7e-11e7-8fb7-56f10f9d1418.png">

5.update "lb" service with a new port 33:33
<img width="1252" alt="screen shot 2017-09-18 at 2 34 16 pm" src="https://user-images.githubusercontent.com/29208297/30565784-7c6c6bfc-9c7e-11e7-9644-18e7a9ec2bea.png">
we can see that:

- new ports 33:33 being allocated with ip belongs to "default2" host

- oldPorts (11:11, and 22:22) still being bounded with their previous allocated ip 
<img width="565" alt="screen shot 2017-09-18 at 2 36 11 pm" src="https://user-images.githubusercontent.com/29208297/30565881-be6020da-9c7e-11e7-9ea6-bf1716edc720.png">
same for API and mysql database:

for API:
```
ports": [ 3 items
"192.168.30.30:11:11/tcp",
"192.168.30.30:22:22/tcp",
"192.168.30.30:33:33/tcp",
],
```

for mysql DB:
```
mysql> select id, name, public_port, data, state from port where removed is null;
+-----+------+-------------+--------------------------------------------+-----------+
| id  | name | public_port | data                                       | state     |
+-----+------+-------------+--------------------------------------------+-----------+
| 523 | NULL |          11 | {"fields":{"bindAddress":"192.168.30.30"}} | active    |
| 524 | NULL |          22 | {"fields":{"bindAddress":"192.168.30.30"}} | active    |
| 525 | NULL |          33 | {"fields":{"bindAddress":"192.168.30.30"}} | active    |
+-----+------+-------------+--------------------------------------------+-----------+
```

**Comment by csnavely** [comment]
Nice!

**Comment by asrivats15** [comment]
We are currently experiencing load balancer issues.  When using 3 Ip’s on a single host (2 VIP’s) and use the same port on two different IPs, I am getting an error:
`Expected state running but got stopped: Allocation failed: Error reserving resources: [10.23.168.29:8443:8443/tcp, 10.23.168.29:443:443/tcp portReservation, instanceReservation: 1
`
Added the IPs to scheduler IP list on the host as such:
`System    io.rancher.scheduler.ips   10.23.168.30, 10.23.168.32, 10.23.168.29
`
This issue also seems intermittent.  The initial setup with the load balancer with a port seems to work.  However any restarts or changes then makes it impossible to reuse the same port as stated above.  Please advise, thanks.

**Comment by cjellick** [comment]
@asrivats15 can you open a separate issue for the problem you are experiencing?

**Comment by csnavely** [comment]
Hi, I just wanted to check if this fix was targeted for a release yet. I've confirmed it's not present in the 1.6.10 release we just installed.

**Comment by cjellick** [comment]
Yes, targeted for november.
FYI, the "resolved" label doesnt mean its in a release. If the issue is open, it isn't in a release yet.

**Comment by csnavely** [comment]
Sounds good; thanks!

**Comment by sangeethah** [comment]
Able to reproduce this issue with 1.6.10 setup having hosts with scheduler Ips.
Create a Lb service with 2 listening ports - 7777 and 8888 
At this point , the LB ports get bound to the scheduler Ips as expected.
Edit LB and add a new listening port say 9999.
After this , all the existing ports ( 7777 and 8888) and new port (9999) gets bound to all interfaces.

```
root@ip-172-31-20-57:/home/ubuntu# iptables-save | grep 7777
-A CATTLE_RAW_PREROUTING -d 172.31.20.57/32 ! -i docker0 -p tcp -m tcp --dport 7777 -j MARK --set-xmark 0x1068/0xffffffff
-A CATTLE_HOSTPORTS_POSTROUTING -s 10.42.100.25/32 -d 10.42.100.25/32 -p tcp -m tcp --dport 7777 -j MASQUERADE
-A CATTLE_OUTPUT -p tcp -m tcp --dport 7777 -m addrtype --dst-type LOCAL -j DNAT --to-destination 10.42.100.25:7777
-A CATTLE_PREROUTING -d 172.31.20.57/32 ! -i docker0 -p tcp -m tcp --dport 7777 -j DNAT --to-destination 10.42.100.25:7777
-A CATTLE_PREROUTING -d 172.31.20.57/32 -p tcp -m tcp --dport 7777 -j DNAT --to-destination 10.42.100.25:7777
root@ip-172-31-20-57:/home/ubuntu# iptables-save | grep 8888
-A CATTLE_RAW_PREROUTING -d 172.31.20.57/32 ! -i docker0 -p tcp -m tcp --dport 8888 -j MARK --set-xmark 0x1068/0xffffffff
-A CATTLE_HOSTPORTS_POSTROUTING -s 10.42.100.25/32 -d 10.42.100.25/32 -p tcp -m tcp --dport 8888 -j MASQUERADE
-A CATTLE_OUTPUT -p tcp -m tcp --dport 8888 -m addrtype --dst-type LOCAL -j DNAT --to-destination 10.42.100.25:8888
-A CATTLE_PREROUTING -d 172.31.20.57/32 ! -i docker0 -p tcp -m tcp --dport 8888 -j DNAT --to-destination 10.42.100.25:8888
-A CATTLE_PREROUTING -d 172.31.20.57/32 -p tcp -m tcp --dport 8888 -j DNAT --to-destination 10.42.100.25:8888
root@ip-172-31-20-57:/home/ubuntu# iptables-save | grep 8888
-A CATTLE_RAW_PREROUTING ! -i docker0 -p tcp -m tcp --dport 8888 -j MARK --set-xmark 0x1068/0xffffffff
-A CATTLE_HOSTPORTS_POSTROUTING -s 10.42.100.25/32 -d 10.42.100.25/32 -p tcp -m tcp --dport 8888 -j MASQUERADE
-A CATTLE_OUTPUT -p tcp -m tcp --dport 8888 -m addrtype --dst-type LOCAL -j DNAT --to-destination 10.42.100.25:8888
-A CATTLE_PREROUTING ! -i docker0 -p tcp -m tcp --dport 8888 -j DNAT --to-destination 10.42.100.25:8888
-A CATTLE_PREROUTING -p tcp -m tcp --dport 8888 -m addrtype --dst-type LOCAL -j DNAT --to-destination 10.42.100.25:8888

After Lb update :

root@ip-172-31-20-57:/home/ubuntu# iptables-save | grep 7777
-A CATTLE_RAW_PREROUTING ! -i docker0 -p tcp -m tcp --dport 7777 -j MARK --set-xmark 0x1068/0xffffffff
-A CATTLE_HOSTPORTS_POSTROUTING -s 10.42.100.25/32 -d 10.42.100.25/32 -p tcp -m tcp --dport 7777 -j MASQUERADE
-A CATTLE_OUTPUT -p tcp -m tcp --dport 7777 -m addrtype --dst-type LOCAL -j DNAT --to-destination 10.42.100.25:7777
-A CATTLE_PREROUTING ! -i docker0 -p tcp -m tcp --dport 7777 -j DNAT --to-destination 10.42.100.25:7777
-A CATTLE_PREROUTING -p tcp -m tcp --dport 7777 -m addrtype --dst-type LOCAL -j DNAT --to-destination 10.42.100.25:7777
root@ip-172-31-20-57:/home/ubuntu# iptables-save | grep 9999
-A CATTLE_RAW_PREROUTING ! -i docker0 -p tcp -m tcp --dport 9999 -j MARK --set-xmark 0x1068/0xffffffff
-A CATTLE_HOSTPORTS_POSTROUTING -s 10.42.100.25/32 -d 10.42.100.25/32 -p tcp -m tcp --dport 9999 -j MASQUERADE
-A CATTLE_OUTPUT -p tcp -m tcp --dport 9999 -m addrtype --dst-type LOCAL -j DNAT --to-destination 10.42.100.25:9999
-A CATTLE_PREROUTING ! -i docker0 -p tcp -m tcp --dport 9999 -j DNAT --to-destination 10.42.100.25:9999
-A CATTLE_PREROUTING -p tcp -m tcp --dport 9999 -m addrtype --dst-type LOCAL -j DNAT --to-destination 10.42.100.25:9999
root@ip-172-31-20-57:/home/ubuntu# 
```

**Comment by sangeethah** [comment]
Tested with 1.6.11-rc3 setup having hosts with scheduler Ips.
Create a Lb service with 1 listening ports .
At this point , the LB ports get bound to the scheduler Ips as expected.
Edit LB and add a new listening port to LB.
After this , existing port and new port are bound to the scheduler Ips as expected.

@csnavely , this fix will be available in the 1.6.11 release.
